src/include/cpu/rpp_cpu_simd.hpp

```diff
@@ -1439,6 +1669,128 @@ static inline __m128 atan2_ps( __m128 y, __m128 x )
     return result;
 }
 
+static inline __m256 log_ps(__m256 x)
+{
+    __m256 e;
+    __m256i emm0;
+    __m256 one = *(__m256 *)&_ps_1;
+    __m256 invalid_mask = _mm256_cmp_ps(x, avx_p0, _CMP_LE_OQ);
+
+    // cut off denormalized stuff
+    x = _mm256_max_ps(x, *(__m256 *)&_ps_min_norm_pos_avx);
+
+    // part 1: x = frexpf(x, &e);
+    emm0 = _mm256_srli_epi32(_mm256_castps_si256(x), 23);
+
+    // keep only the fractional part
+    x = _mm256_and_ps(x, *(__m256 *)&_ps_inv_mant_mask_avx);
+    x = _mm256_or_ps(x, *(__m256 *)&_ps_0p5_avx);
+
+    emm0 = _mm256_sub_epi32(emm0, *(__m256i *)&_pi32_0x7f_avx);
+    e = _mm256_cvtepi32_ps(emm0);
+
+    e = _mm256_add_ps(e, one);
+
+    // part 2: if( x < SQRTHF ) { e -= 1; x = x + x - 1.0; } else { x = x - 1.0; }
+    __m256 z, y;
+    __m256 mask = _mm256_cmp_ps(x, *(__m256 *)&xmm_cephesSQRTHF, _CMP_LT_OQ);
+    __m256 tmp = _mm256_and_ps(x, mask);
+    x = _mm256_sub_ps(x, one);
+    e = _mm256_sub_ps(e, _mm256_and_ps(one, mask));
+    x = _mm256_add_ps(x, tmp);
+    z = _mm256_mul_ps(x,x);
+    y = *(__m256 *)&avx_cephesLogP0;
+    y = _mm256_mul_ps(y, x);
+    y = _mm256_add_ps(y, *(__m256 *)&avx_cephesLogP1);
+    y = _mm256_mul_ps(y, x);
+    y = _mm256_add_ps(y, *(__m256 *)&avx_cephesLogP2);
+    y = _mm256_mul_ps(y, x);
+    y = _mm256_add_ps(y, *(__m256 *)&avx_cephesLogP3);
+    y = _mm256_mul_ps(y, x);
+    y = _mm256_add_ps(y, *(__m256 *)&avx_cephesLogP4);
+    y = _mm256_mul_ps(y, x);
+    y = _mm256_add_ps(y, *(__m256 *)&avx_cephesLogP5);
+    y = _mm256_mul_ps(y, x);
+    y = _mm256_add_ps(y, *(__m256 *)&avx_cephesLogP6);
+    y = _mm256_mul_ps(y, x);
+    y = _mm256_add_ps(y, *(__m256 *)&avx_cephesLogP7);
+    y = _mm256_mul_ps(y, x);
+    y = _mm256_add_ps(y, *(__m256 *)&avx_cephesLogP8);
+    y = _mm256_mul_ps(y, x);
+    y = _mm256_mul_ps(y, z);
+    tmp = _mm256_mul_ps(e, *(__m256 *)&avx_cephesLogQ1);
+    y = _mm256_add_ps(y, tmp);
+    tmp = _mm256_mul_ps(z, *(__m256 *)&_ps_0p5_avx);
+    y = _mm256_sub_ps(y, tmp);
+    tmp = _mm256_mul_ps(e, *(__m256 *)&avx_cephesLogQ2);
+    x = _mm256_add_ps(x, y);
+    x = _mm256_add_ps(x, tmp);
+    x = _mm256_or_ps(x, invalid_mask); // negative arg will be NAN
+
+    return x;
+}
+
+static inline __m128 log_ps(__m128 x)
```

Comment:
Is there is reference link to the algorithm here? Please include. Otherwise insert comment on what this is doing

Reply:
SSE reference added - https://github.com/RJVB/sse_mathfun/blob/master/sse_mathfun.h